pkg/ddl/ddl.go

```diff
@@ -560,7 +561,31 @@ func (d *ddl) RegisterStatsHandle(h *handle.Handle) {
 
 // asyncNotifyEvent will notify the ddl event to outside world, say statistic handle. When the channel is full, we may
 // give up notify and log it.
-func asyncNotifyEvent(jobCtx *jobContext, e *notifier.SchemaChangeEvent, job *model.Job) {
+func asyncNotifyEvent(jobCtx *jobContext, e *notifier.SchemaChangeEvent, job *model.Job, sessPool *sess.Pool) {
+	if intest.InTest {
+		if sessPool == nil || notifier.DefaultStore == nil {
+			return
+		}
+		// skip notify for system databases, system databases are expected to change at
+		// bootstrap and other nodes can also handle the changing in its bootstrap rather
+		// than be notified.
+		if tidbutil.IsMemOrSysDB(job.SchemaName) {
+			return
+		}
+		sessCtx, err := sessPool.Get()
```

Comment:
we must use the same session of worker, to commit the change in the same transaction of commiting the DDL. Here you just choose a unused session from session pool.

Reply:
update In [cdecc96](https://github.com/pingcap/tidb/pull/56425/commits/cdecc96d5fb856cb15eabcb2117009d9ba9bd0fe)